integrationtests/tests/tests.rs

```diff
@@ -1203,3 +1207,35 @@ async fn cannot_replay_transactions() -> Result<()> {
     })
     .await
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn verify_dummy_migration() -> Result<()> {
+    let fed_db = |decoders| async {
+        // Insert a bunch of old dummy data that needs to be migrated to a new version
+        let db = Database::new(MemDatabase::new(), decoders);
+        let dbtx = db.begin_transaction().await;
+        let mut module_dbtx = dbtx.new_module_tx(LEGACY_HARDCODED_INSTANCE_ID_DUMMY);
```

Comment:
I'm not super crazy about re-adding the LEGACY_HARDCODED_INSTANCE_ID, even if its just for the dummy module. But I couldn't figure out a way to "inject" data into the db before the federation has started without adding specialized logic into the `Fixtures`, which didn't seem like the right solution.

Reply:
I'm not actually sure if its possible to not use it, since by definition, when the federation hasn't started yet, the module instance ids haven't been assigned yet.

Reply:
You should not need to add `LEGACY_HARDCODED_INSTANCE_ID`. You can make up any id, as the module under test can't even check it.

Reply:
The issue is that the integration test needs to know which module instance id is assigned to the dummy module, so that it can backfill some test data before the federation starts that will later be migrated when the federation starts. If we don't use the correct ID, then the dummy module won't even see the test data (since it is logically isolated), the migration won't execute and the test will fail.

Reply:
I think kitman just today put module instance ids into `FederationTest`, so we don't have to hardcode them. (grep `mint_id`). Would that help?

Reply:
Unfortunately I don't think it does, since the module instance ids are assigned when the federation is created, and for this test I need to inject dummy data into the database before those ids are assigned so that the migration can be triggered.

Reply:
I think we posted at the same time:

> I think kitman just today put module instance ids into FederationTest, so we don't have to hardcode them. (grep mint_id). Would that help?

Reply:
Removed the integration test and created an equivalent unit test instead